dashboard/modules/job/cli_utils.py

```diff
@@ -0,0 +1,46 @@
+from typing import Union
+
+import click
+import functools
+
+
+def bool_cast(string: str) -> Union[bool, str]:
+    """Cast a string to a boolean if possible, otherwise return the string."""
+    if string.lower() == "true" or string == "1":
+        return True
+    elif string.lower() == "false" or string == "0":
+        return False
+    else:
+        return string
+
+
+class BoolOrStringParam(click.ParamType):
+    """A click parameter that can be either a boolean or a string."""
+
+    name = "BOOL | TEXT"
+
+    def convert(self, value, param, ctx):
+        if isinstance(value, bool):
```

Comment:
Out of curiosity how could a bool (as opposed to a string "true") appear here? Isn't everything passed in the command line a string?

Reply:
Answered in the click doc

> Values from user input or the command line will be strings, but default values and Python arguments may already be the correct type. The custom type should check at the top if the value is already valid and pass it through to support those cases.